ansible_collections/arista/avd/roles/eos_designs/templates/base/base.py

```diff
@@ -0,0 +1,658 @@
+                'routing': True
+            }
+        }
+        if get(self._switch, 'evpn_multicast') is True:
+            router_multicast['ipv4']['software_forwarding'] = 'sfe'
+        return router_multicast
+
+    @cached_property
+    def hardware_counters(self):
+        """
+        hardware_counters set based on hardware_counters.features variable
+        """
+        if (hardware_counter_features := get(self._hostvars, 'hardware_counters.features')) is None:
+            return None
+        features = [{feature: direction} for feature, direction in hardware_counter_features.items()]
+        return {"features": features}
+
+    @cached_property
+    def hardware(self):
+        """
+        hardware set based on platform_speed_groups variable and switch.platform fact.
+        Converting nested dict to list of dict to support avd_v4.0
+        """
+        if (platform_speed_groups := get(self._hostvars, 'platform_speed_groups')) is None:
+            return None
+
+        tmp_speed_groups = {}
+        if (switch_platform := get(self._switch, 'platform')) is not None:
+            # converting nested dict to list of dict to support avd_v4.0
+            platform_speed_groups = self._convert_dicts(platform_speed_groups, 'platform', 'speeds')
+            for platform_item in platform_speed_groups:
+                if platform_item['platform'] == switch_platform:
+                    # converting nested dict to list of dict to support avd_v4.0
+                    speeds = self._convert_dicts(platform_item.get('speeds'), 'speed', 'speed_groups')
+                    for speed in self._natural_sort(speeds, 'speed'):
+                        for speed_group in speed['speed_groups']:
+                            tmp_speed_groups[speed_group] = speed['speed']
+        if tmp_speed_groups:
+            hardware = {'speed_groups': {}}
+            for speed_group in self._natural_sort(tmp_speed_groups):
+                hardware['speed_groups'][speed_group] = {'serdes': tmp_speed_groups[speed_group]}
+            return hardware
+
+    @cached_property
+    def daemon_terminattr(self):
+        """
+        daemon_terminattr set based on cvp_instance_ip and cvp_instance_ips variables
+
+        Updating cvaddrs and cvauth considering conditions for cvaas and cvp_on_prem IPs
+
+            if 'arista.io' in cvp_instance_ip:
+                 <updating as cvaas_ip>
+            else:
+                 <updating as cvp_on_prem ip>
+        """
+        cvp_instance_ip = get(self._hostvars, 'cvp_instance_ip')
+        cvp_instance_ips = get(self._hostvars, 'cvp_instance_ips')
+        if cvp_instance_ip is None and cvp_instance_ips is None:
+            return None
+        cvp_instance_ip_list = []
+        if cvp_instance_ip is not None:
+            cvp_instance_ip_list.append(cvp_instance_ip)
+        elif cvp_instance_ips is not None:
+            for cvp_ip in cvp_instance_ips:
+                cvp_instance_ip_list.append(cvp_ip)
+
```

Comment:
This can be refactored and simplified a bit:
```suggestion
        cvp_instance_ip = get(self._hostvars, 'cvp_instance_ip')
        cvp_instance_ip_list = get(self._hostvars, 'cvp_instance_ips', [])
        if cvp_instance_ip is not None:
            cvp_instance_ip_list.append(cvp_instance_ip)

        if not cvp_instance_ip_list:
            return None   
```
And then remove the if check below, since it is included here.